pkg/bindings/connection.go

```diff
@@ -108,23 +111,76 @@ func NewConnectionWithIdentity(ctx context.Context, uri string, identity string,
 	var connection Connection
 	switch _url.Scheme {
 	case "ssh":
+		userinfo := _url.User
+		if _url.User == nil {
+			u, err := user.Current()
+			if err != nil {
+				return nil, err
```

Comment:
Can you wrap this one? I feel like our errors out of machine are rarely good, so I'd like to at least try to be better in new code